concent_api/core/validation.py

```diff
@@ -218,3 +225,81 @@ def validate_list_task_to_compute_ids(subtask_results_accepted_list):
+    if file_transfer_token.storage_cluster_address != settings.STORAGE_CLUSTER_ADDRESS:
+        return ('This token does not allow file transfers to/from the cluster you are trying to access.', ErrorCode.MESSAGE_STORAGE_CLUSTER_WRONG_CLUSTER)
+
+    # -CLIENT_PUBLIC_KEY
+    if not isinstance(file_transfer_token.authorized_client_public_key, bytes):
+        return ('Wrong type of authorized_client_public_key field value.', ErrorCode.MESSAGE_AUTHORIZED_CLIENT_PUBLIC_KEY_WRONG_TYPE)
+    # -FILES
+    if not all(isinstance(file, dict) for file in file_transfer_token.files):
+        return ('Wrong type of files field value.', ErrorCode.MESSAGE_FILES_WRONG_TYPE)
+    transfer_token_paths_to_files = [file["path"] for file in file_transfer_token.files]
+    if len(transfer_token_paths_to_files) != len(set(transfer_token_paths_to_files)):
+        return ('File paths in the token must be unique', ErrorCode.MESSAGE_FILES_PATHS_NOT_UNIQUE)
+
+    if not all(isinstance(file["checksum"], str) for file in file_transfer_token.files):
+        return ("'checksum' must be a string.", ErrorCode.MESSAGE_FILES_CHECKSUM_WRONG_TYPE)
+    if any((len(file["checksum"]) == 0 or file["checksum"].isspace()) for file in file_transfer_token.files):
+        return ("'checksum' cannot be blank or contain only whitespace.", ErrorCode.MESSAGE_FILES_CHECKSUM_EMPTY)
+    if not all(":" in file["checksum"] for file in file_transfer_token.files):
+        return ("'checksum' must consist of two parts separated with a semicolon.", ErrorCode.MESSAGE_FILES_CHECKSUM_WRONG_FORMAT)
+    file_checksums = [tuple(file["checksum"].split(":")) for file in file_transfer_token.files]
+    if not set(checksum_type for checksum_type, checksum in file_checksums).issubset(set(HashingAlgorithm._value2member_map_)):  # type: ignore
+        return ("One of the checksums is from an unsupported hashing algorithm.", ErrorCode.MESSAGE_FILES_CHECKSUM_INVALID_ALGORITHM)
+
+    assert set(HashingAlgorithm) == {HashingAlgorithm.SHA1}, "If you add a new hashing algorithms, you need to add validations below."
+    if any(VALID_SHA1_HASH_REGEX.fullmatch(file_checksum[1]) is None for file_checksum in file_checksums):
+        return ("Invalid SHA1 hash.", ErrorCode.MESSAGE_FILES_CHECKSUM_INVALID_SHA1_HASH)
+
+    file_sizes = [file["size"] for file in file_transfer_token.files]
+    if any((file_size is None) for file_size in file_sizes):
+        return ("'size' must be an integer.", ErrorCode.MESSAGE_FILES_SIZE_EMPTY)
+    for file_size in file_sizes:
+        try:
+            int(file_size)
+        except (ValueError, TypeError):
+            return ("'size' must be an integer.", ErrorCode.MESSAGE_FILES_SIZE_WRONG_TYPE)
+
+    if any(int(file["size"]) < 0 for file in file_transfer_token.files):
+        return ("'size' must not be negative.", ErrorCode.MESSAGE_FILES_SIZE_NEGATIVE)
+
+    # Validate category in FileInfo
+    if not all('category' in file for file in file_transfer_token.files):
```

Comment:
Same here for combining into single loop for `category` check.

Reply:
I'm not sure if those "loopable" category checks are even necessary? Checks for `FileTransferToken.FileInfo()` are done by golem_messages, so we can't even create test for missing 'category' field or to put something else in there then predetermined Enums.

Reply:
Those checks were required in https://github.com/golemfactory/concent/issues/355. Maybe they can be dropped now, @cameel what do you think ?

Reply:
Done. I've switched existence and enum checks to `asserts`. Only one our custom checks is left, so no loop is needed.

Reply:
@cameel Ok but the code is more readable and optimal now, and I see @Jakub89 already did this so I let you guys decide. For me both are ok.